src/EnergyPlus/FileSystem.hh

```diff
@@ -85,17 +88,45 @@ namespace FileSystem {
 
     extern std::string const exeExtension;
 
-    enum class InputFileType
+    enum class FileTypes
     {
+        Unknown,
+        // JSON types should go first,
         EpJSON,
-        IDF,
+        GLHE,
+        JSON,
+        last_json_type = JSON,
         CBOR,
         MsgPack,
         UBJSON,
         BSON,
-        Unknown
+        last_binary_json_type = BSON,
+        CSV,
+        TSV,
+        TXT,
+        ESO,
+        MTR,
+        IMF,
+        IDF,
+        last_flat_file_type = IDF
     };
 
+    inline constexpr bool is_all_json_type(FileTypes t) {
+        return t > FileTypes::Unknown && t <= FileTypes::last_binary_json_type;
+    }
+    inline constexpr bool is_json_type(FileTypes t) {
+      return t > FileTypes::Unknown && t <= FileTypes::last_json_type;
+    }
+    inline constexpr bool is_binary_json_type(FileTypes t) {
+        return t > FileTypes::last_json_type && t <= FileTypes::last_binary_json_type;
+    }
+    inline constexpr bool is_idf_type(FileTypes t) {
+        return t == FileTypes::IDF || t == FileTypes::IMF;
```

Comment:
Not a giant deal because this will not be called often, but can use the BITF trick.

Reply:
Yeah this isn't called often and when these functions are called often, they are part of template functions and thus fully compile time calculations.